types/src/dkg/real_dkg/mod.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    dkg::{real_dkg::rounding::DKGRounding, DKGSessionMetadata, DKGTrait},
+    validator_verifier::{ValidatorConsensusInfo, ValidatorVerifier},
+};
+use anyhow::{anyhow, ensure};
+use aptos_crypto::{bls12381, bls12381::PrivateKey};
+use aptos_dkg::{
+    pvss,
+    pvss::{
+        traits::{Convert, Reconstructable, Transcript},
+        Player,
+    },
+};
+use num_traits::Zero;
+use rand::{CryptoRng, RngCore};
+use rounding::{RECONSTRUCT_THRESHOLD, SECRECY_THRESHOLD};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeSet;
+
+pub mod rounding;
+
+pub type WTrx = pvss::das::WeightedTranscript;
+pub type DkgPP = <WTrx as Transcript>::PublicParameters;
+pub type SSConfig = <WTrx as Transcript>::SecretSharingConfig;
+pub type EncPK = <WTrx as Transcript>::EncryptPubKey;
+
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq)]
+pub struct DKGPvssConfig {
+    pub epoch: u64,
+    // weighted config for randomness generation
+    pub wconfig: SSConfig,
+    // DKG public parameters
+    pub pp: DkgPP,
+    // DKG encryption public keys
+    pub eks: Vec<EncPK>,
+}
+
+impl DKGPvssConfig {
+    pub fn new(epoch: u64, wconfig: SSConfig, pp: DkgPP, eks: Vec<EncPK>) -> Self {
+        Self {
+            epoch,
+            wconfig,
+            pp,
+            eks,
+        }
+    }
+}
+
+pub fn build_dkg_pvss_config(
+    cur_epoch: u64,
+    next_validators: &[ValidatorConsensusInfo],
+) -> DKGPvssConfig {
+    let validator_stakes: Vec<u64> = next_validators.iter().map(|vi| vi.voting_power).collect();
+
+    let dkg_rounding =
+        DKGRounding::new(&validator_stakes, SECRECY_THRESHOLD, RECONSTRUCT_THRESHOLD);
+
+    println!(
+        "[Randomness] rounding: epoch {} starts, profile = {:?}",
+        cur_epoch, dkg_rounding.profile
+    );
+
+    let validator_consensus_keys: Vec<bls12381::PublicKey> = next_validators
+        .iter()
+        .map(|vi| vi.public_key.clone())
+        .collect();
+
+    let consensus_keys: Vec<EncPK> = validator_consensus_keys
+        .iter()
+        .map(|k| k.to_bytes().as_slice().try_into().unwrap())
+        .collect::<Vec<_>>();
+
+    let wconfig = dkg_rounding.wconfig.clone();
+
+    let pp = DkgPP::default_with_bls_base();
+
+    DKGPvssConfig::new(cur_epoch, wconfig.clone(), pp, consensus_keys)
+}
+
+#[derive(Debug)]
+pub struct RealDKG {}
```

Comment:
let's not call it RealDKG, how about wDKG or something?

Reply:
will rename later!